src/components/ComputeModal.js

```diff
@@ -1157,9 +1197,11 @@ export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDis
       value: selectedLeoImage,
       onChange: ({ value }) => {
         const requiresSpark = _.find({ image: value }, leoImages)?.requiresSpark
+        const newSparkMode = requiresSpark ? (sparkMode || 'master') : false
```

Comment:
It seems preferable to avoid multi-typed variables wherever possible despite the language supporting it. I don't like that this variable is overloaded with both the spark mode, and a boolean which I don't inherently understand the meaning of. 

I'd say either change `false` to `undefined` (which would be a more clear falsey value to indicate its not applicable), or separate it into a flag that is checked before the (enum?) string